src/pymovements/gaze/io.py

```diff
@@ -360,6 +362,98 @@ def from_asc(
             for fileinfo_key, fileinfo_dtype in column_schema_overrides.items()
         ])
 
+    if experiment is None:
+        experiment = Experiment(sampling_rate=metadata['sampling_rate'])
+    if experiment.screen is None:
```

Comment:
Actually, `Experiment.screen` can never be `None` until someone tampered with it. That's why this line isn't covered and you would need to create a bit of an artificial test where you tamper with the `screen` value.

`Experiment.screen` is set right at the start of `Experiment.__init__(...)`:

https://github.com/aeye-lab/pymovements/blob/4042879f4577b68b08c197095ae1a413c332311c/src/pymovements/gaze/experiment.py#L98-L105

Instead I think it's probably best to remove this if clause completely and handle the edge case in a follow-up PR: the `Experiment` class can be improved by implementing the `screen` field as a property. This way we can guarantee its integrity (being of type `Screen`).